controllers/argocd/deployment.go

```diff
@@ -1056,6 +1056,10 @@ func (r *ReconcileArgoCD) reconcileRepoDeployment(cr *argoprojv1a1.ArgoCD) error
 				deploy.Spec.Template.Spec.Containers[1:]...)
 			changed = true
 		}
+		if !reflect.DeepEqual(deploy.Spec.Template.Spec.InitContainers, existing.Spec.Template.Spec.InitContainers) {
```

Comment:
The desired InitContainers are in `deploy.Spec.Template.Spec.InitContainers`. Doing this here should be enough, right?  Do we need line 1063 - 1080 (copying volumne monnts, env, resources, command to the first initContainer)?